introduction.Rmd

```diff
@@ -9,27 +9,185 @@ library("dplyr")
+
+Tidy evaluation is a set of concepts and tools that make it possible to use tidyverse grammars when columns are specified indirectly. In particular, you will need to learn some tidy eval to extract a tidyverse pipeline in a reusable function.
+
+The first chapter [Why and how](#why-how) provides the motivation for tidy eval, presents the problems that it poses in day-to-day programming, and the general theory and tools for solving those. If you are in a hurry, you can jump straight to [Do you need tidy eval?](do-you-need). A lot can be done without writing a single line of tidy eval! If you are positive you need it to solve your problem, [Getting up to speed](#up-to-speed) is a self-contained chapter that will teach you the basic workflow of wrapping a tidyverse pipeline in a reusable function. The book ends with a series of recipes and idioms for solving various dplyr and ggplot2 problems.
+
+
+# Why and how {#why-how}
+
+Tidy evaluation is a framework for metaprogramming in R, used throughout the tidyverse to implement data masking. Metaprogramming is about using a programming language to manipulate or modify its own code. This idea is used throughout the tidyverse to change the context of computation of certain pieces of R code.
+
+Changing the context of evaluation is useful for three main purposes:
+
+* To promote data frames to **full blown scopes**, where columns are exposed as named objects.
+
+* To execute your R code in a **foreign environment**. For instance, dbplyr translates ordinary dplyr pipelines to SQL queries.
+
+* To implement **special rules** for ordinary R operators. For instance, selection functions such as `dplyr::select()` or `tidyr::gather()` implement specific behaviours for `c()`, `:` and `-`.
+
+
+## Data masking
+
+Of these goals, the promotion of data frames is the most important because data is often the most relevant context for data analysts. We believe that R and the tidyverse are [human-centered](https://principles.tidyverse.org/unifying-principles.html#human-centered) in big part because the data frame is available for direct use in computations, without syntax and boilerplate getting in the way. Formulas for statistical models are a prime example of human-centered syntax in R. Data masking and special operator rules make formulas an intuitive interface for model specification.
+
+When the contents of the data frame are temporarily promoted as first class objects, we say the data **masks** the workspace:
 
 ```{r, eval = FALSE}
 library("dplyr")
 
-# Masking data
-starwars %>%
-  filter(
-    height < 200,
-    gender == "male"
-  )
+starwars %>% filter(
+  height < 200,
+  gender == "male"
+)
+```
+
+Compare to the equivalent subsetting code where it is necessary to be explicit about where the columns come from:
 
-# Qualified data
-starwars[starwars$height < 200 & starwars$gender == "male", ]
+```{r, eval = FALSE}
+index <-
+  !is.na(starwars$height) & !is.na(starwars$gender) &
```

Comment:
I think you should remove the `!is.na()` stuff because that's related to `filter()` not to masking (or at least move it to a footnote)